tracer/src/Datadog.Trace/Agent/Api.cs

```diff
@@ -20,29 +21,36 @@ internal class Api : IApi
     {
         private const string TracesPath = "/v0.4/traces";
 
-        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<Api>();
+        private static readonly IDatadogLogger StaticLog = DatadogLogging.GetLoggerFor<Api>();
 
+        private readonly IDatadogLogger _log;
         private readonly IApiRequestFactory _apiRequestFactory;
         private readonly IDogStatsd _statsd;
         private readonly string _containerId;
         private readonly Uri _tracesEndpoint;
-        private readonly IDatadogTracer _tracer;
+        private readonly ISampler _sampler;
+        private readonly bool _isPartialFlushEnabled;
         private string _cachedResponse;
-
-        public Api(Uri baseEndpoint, IApiRequestFactory apiRequestFactory, IDogStatsd statsd)
-            : this(baseEndpoint, apiRequestFactory, statsd, tracer: null)
+        private string _agentVersion;
+
+        public Api(
+            Uri baseEndpoint,
+            IApiRequestFactory
+            apiRequestFactory,
+            IDogStatsd statsd,
+            ISampler sampler,
```

Comment:
What are your thoughts on passing a callback into `Api` instead of the sampler itself? Separation of concerns and all that. Looks like we could pass an `Action<Dictionary<string, float>>` here since we only use it for `_sampler.SetDefaultSampleRates()` below.

Reply:
Yeah, I considered that, but decided this was the smaller change for now. Certainly happy to take that route, but maybe in a follow-up PR?

Reply:
If we're to refactor that, then I think it makes more sense to squeeze it in the return value of `SendTracesAsync` rather than using a callback

Reply:
> If we're to refactor that, then I think it makes more sense to squeeze it in the return value of SendTracesAsync rather than using a callback

Not sure I agree with you here - as that just shifts the problem - we would have to start injecting `ISampler` into the `AgentWriter` (rather than `API`), which doesn't really improve separation, and also move the caching and response deserialization logic out of the `API` into the `AgentWriter`, which arguably makes it worse. I think the callback approach is probably a better option?